### DISCUSSION TITLE
Adding user_float values to header

### DISCUSSION THREAD

**Discussion by saranevessilva** [gadgetron/gadgetron#1164] [Q&A]
Hi all, I'm trying to store some information in the user_float parameters in the reconstructed image header using the from_array function from ismrmrd/image.py (https://github.com/ismrmrd/ismrmrd-python/blob/v1.9.5/ismrmrd/image.py). However, when I tested this in the scanner most parameters in the header of the .h5 file were empty apart from version, data_type, channels, and matrix_size. It seems that in this function only these four parameters are actually stored in the header, as seen in the screen shot.
![Screenshot 2022-10-04 115343](https://user-images.githubusercontent.com/72754856/193801573-d389c4bf-d463-4356-aacd-f51c92815f50.png)

I tried to add user_float to the image_properties but it seems they need some sort of conversion/serialisation. I also tried to just use the from_acquisition function in ImageHeader class to simply store a header file but get the error- TypeError: No appropriate writer found for item of type '<class 'ismrmrd.image.ImageHeader'>'.

Has anyone done this before?

**Comment by hansenms** [comment]
It is a bit unclear from this what you are actually trying. On the scanner there is not .h5 file, right? In the code you list above, it calls `ImageHeader.from_acquisition`, which does, I think transfer the user floats: https://github.com/ismrmrd/ismrmrd-python/blob/master/ismrmrd/image.py#L74. I might need a bit more information to be able to help.

**Reply by saranevessilva** [reply]
in my pipeline I have a Python-based Gadget where I apply a neural network to the reconstructed image and extract some motion parameters that need to be stored in the image header (I use the empty user_float parameters for storage) and sent to the ICE chain with the reconstructed image. In order to store information in the header, I followed an example available here https://github.com/gadgetron/GadgetronOnlineClass/blob/master/Courses/Day1/Lecture3/Part2/correction/my_first_buffered_data_python_gadget.py:
![Screenshot 2022-10-06 173236](https://user-images.githubusercontent.com/72754856/194368820-2521c4ec-5626-4304-93e2-3293929a0229.png)
Even though the information I intended to save in the header was allocated to the user floats (as in the screenshot), when I tried to print them out in the IceGadgetron program in the scanner these parameters were empty. I then used this script https://github.com/roopchansinghv/ismrmrd_to_nifti/blob/master/img_h5_to_nii/python/read_and_convert_ismrmrd.ipynb to load my reconstructed image file (.h5) and only the parameters version, data_type, channels, and matrix_size had information stored- exactly the same parameters that I managed to print information from in the scanner. Basically the way I worked around this was by modifying the from_array function from image.py and add the parameters I want to store in the header as members of this function: 
![Screenshot 2022-10-06 174735](https://user-images.githubusercontent.com/72754856/194371826-d9194b5e-0998-4741-8e7f-2dae80f3d1b0.png)
![Screenshot 2022-10-06 174816](https://user-images.githubusercontent.com/72754856/194371984-1379e7eb-7ee3-4125-841e-1d78a2a145e6.png)
and also added the "user_floats" key to image_properties, alongside the array that contains the motion information. This worked! :)
